Integrations/integration-CounterTack.yml

```diff
@@ -0,0 +1,1716 @@
+description: CounterTack empowers endpoint security teams to assure endpoint protection
+  for Identifying Cyber Threats. Integrating a predictive endpoint protection platform
+configuration:
+  - display: Server URL
+    name: server
+    defaultvalue: https://democloud.countertack.com
+    type: 0
+    required: true
+  - display: User Name
+    name: credentials
+    defaultvalue: ""
+    type: 9
+    required: true
+  - display: Trust any certificate (unsecure)
+    name: insecure
+    defaultvalue: ""
+    type: 8
+    required: false
+  - display: Fetch incidents
+    name: isFetch
+    defaultvalue: ""
+    type: 8
+    required: false
+  - display: Incident type
+    name: incidentType
+    defaultvalue: ""
+    type: 13
+    required: false
+  - display: Use system proxy settings
+    name: proxy
+    defaultvalue: ""
+    type: 8
+    required: false
+script:
+  script: |-
+    ''' IMPORTS '''
+
+    import json
+    import requests
+    import os
+    import base64
+
+    # Disable insecure warnings
+    requests.packages.urllib3.disable_warnings()
+
+    # PREREQUISITES
+    def load_server_url():
+        ''' Cleans and loads the server url from the configuration '''
+        url = demisto.params()['server']
+        url = re.sub('/[\/]+$/', '', url)
+        url = re.sub('\/$', '', url)
+        return url
+
+    # remove proxy if not set to true in params
+    if not demisto.params().get('proxy'):
+        del os.environ['HTTP_PROXY']
+        del os.environ['HTTPS_PROXY']
+        del os.environ['http_proxy']
+        del os.environ['https_proxy']
+
+    ''' GLOBALS/PARAMS '''
+
+    USERNAME = demisto.params().get('credentials').get('identifier')
+    PASSWORD = demisto.params().get('credentials').get('password')
+    SERVER_URL = load_server_url()
+    # Should we use SSL
+    USE_SSL = not demisto.params()['insecure']
+    # Service base URL
+    BASE_PATH = '{}/api/v2'.format(SERVER_URL)
+    # Headers to be sent in requests
+    GET_HEADERS = {
+        'Content-Type': 'application/json'
+    }
+
+
+    def http_request(method, url, headers={}, body=None, body_params=None):
+        """
+        returns the http request
+
+        """
+        #basic authentication
+        try:
+            response = requests.request(
+                method,
+                url,
+                auth=(USERNAME,PASSWORD),
+                headers=headers,
+                verify=USE_SSL,
+                data=body
+        )
+        except requests.exceptions.SSLError as e:
+            LOG(e.message)
+            raise ValueError('An SSL error occurred when trying to connect to the server. Consider configuring unsecure connection in the integration settings.')
+
+        #handle request failure
+        if response.status_code not in range(200,205):
+            message = parse_error_response(response)
+            raise ValueError('Request failed with status code {}\n{}'.format(response.status_code, message))
```

Comment:
use `return_error` instead of raise
rephrase the error message to be something like `Error in API call to CounterTack...`